README.md

```diff
@@ -1085,7 +1089,85 @@ In either case:
     end
     ```
 
+### Separation of Concerns
+
+* <a name="business-logic"></a>Business logic should be in models and
+  services, not controllers or views.<sup>[[link](#business-logic)]</sup>
+
+* <a name="services"></a>Extract service objects to manage complex interactions
+  across models.<sup>[[link](#services)]</sup>
+
+* <a name="poros"></a>In general, extract basic ruby object to wrap complex
+  logic and keep rails objects at a consistent level of abstraction.([More on
+  this][fat-activerecord].)<sup>[[link](#poros)]</sup>
+
+### Persistence
+* <a name="bangy-saves"></a>Prefer `save!`, `create!`, and `update_attributes!`
+  over the non-bangy versions of those methods unless explicitly handling when
+  the save fails.<sup>[[link](#bangy-saves)]</sup>
+
+* <a name="run-validations"></a>Do not skip model validations (with e.g. `inc`
+  or `update_attribute`) unless you're solving a measured performance
+  bottleneck.<sup>[[link](#run-validations)]</sup>
+
+### Testing Strategy
+
+* <a name="integration-tests"></a>Integration test for authentication and correct status
+  codes<sup>[[link](#integration-tests)]</sup>
+
+* <a name="controller-tests"></a>Controller test that appropriate instance
```

Comment:
controller tests should also test flows and paths. for instance, if a user is not logged in and making a request to an api controller, then the status code response should be tested in the controller spec. similarly with success and failure paths. integration tests should test in ideal case condition and an unideal condition to ensure that both cases are handled for the user, but it shouldnt test all the cases because its more time consuming.